src/types/boolobject.rs

```diff
@@ -21,13 +21,13 @@ impl PyBool {
     /// Gets whether this boolean is `true`.
     #[inline]
     pub fn is_true(&self) -> bool {
-        Py2::borrowed_from_gil_ref(&self).is_true()
+        Bound::borrowed_from_gil_ref(&self).is_true()
```

Comment:
`Bound::borrowed_from_gil_ref` feels like quite a mouthful now, I wonder if it would be better to add a new trait method `as_bound` to `PyNativeType` so that this can instead be:

```suggestion
        self.as_bound().is_true()
```

We could similarly also have `PyNativeType::as_borrowed`.

Probably these are better done in a follow-up PR to discuss separately in a smaller diff?

Reply:
Yep, better as a follow-up.

Reply:
(The follow-up PR will also give the opportunity to consider what the public API story for the compatibility methods transitioning between old and new API is.)